src/events/deposits.upsert.service.ts

```diff
@@ -151,4 +165,55 @@ export class DepositsUpsertService {
     }
     return result[0].count;
   }
+
+  async mismatchedDeposits(beforeSequence: number = 0): Promise<
+    (Deposit & { block_main: boolean | null; block_timestamp: Date | null })[]
+  > {
+    const blocksHead = await this.blocksService.head();
+    return await this.prisma.$queryRawUnsafe<
+      (Deposit & { block_main: boolean | null; block_timestamp: Date | null })[]
+    >(
+      `
+      SELECT
+        deposits.*,
+        blocks.main AS block_main,
+        blocks.timestamp AS block_timestamp
+      FROM
+        deposits
+      LEFT JOIN
+        blocks
+      ON blocks.hash = deposits.block_hash
+      WHERE
+        blocks.hash IS NULL OR
+        blocks.main <> deposits.main AND
+        (
+          blocks.hash IS NULL OR
+          blocks.sequence < ${blocksHead.sequence - beforeSequence}
+        )
+      `,
+    );
+  }
+
+  async fixMismatchedDeposits(): Promise<void> {
+    const mismatchedDeposits = await this.mismatchedDeposits(10);
+
+    for (const deposit of mismatchedDeposits) {
+      await this.prisma.$transaction(async (prisma) => {
+        const updatedDeposit = await this.prisma.deposit.update({
+          data: {
+            main: deposit.block_main ?? false,
+          },
+          where: {
+            id: deposit.id,
+          },
+        });
+
+        await this.processDeposit(
```

Comment:
I think this is fine for now, but if we see any starvation or long running worker jobs maybe we can chunk this out down the line.

Reply:
I think it may be worth breaking this loop into separate jobs now -- there are over 80k mismatched deposits right now, so the first time we run this will almost surely be a long running job.